common/captcha/CaptchaSolverInterface.py

```diff
@@ -0,0 +1,90 @@
+from abc import ABC, abstractmethod
+import os
+import cv2
+import time
+
+
+class SolvedCaptcha:
+    """
+    A class interface for representing solved Captchas.
+    """
+
+    def __init__(self, image, text, answer=None, out_dir=None):
+        """
+        :param image: Unprocessed Captcha image
+        :param text: The text contained in the captcha image. (eg: 12+3=)
+        :param answer: The solved captcha's answer (eg: 15). In the 'jumbled text' captchas this is the same as 'text'.
+        :param out_dir: Where to write captchas to upon notify_correct() or notify_incorrect()
+        """
+        self.image = image
+        self.text = text
+        self.answer = answer
+        self.out_dir = out_dir
```

Comment:
Would it make sense to validate and make out_dir now? Seems like it would allow us to reliably avoid the TypeError later and we wouldn't necessarily have to makedirs in the notify functions.

Reply:
The reason I put this in the notify/save functions was to potentially save some unnecessary folder creation in the case the user never calls the notify/save function.

I think it makes more sense to create/check for the existence of the folder only if it's needed.

Reply:
Alrighty. I think this is reasonable in the interface, but may make sense to do it in the constructor of the implementation (since we know it'll be used there).